model/host/db.go

```diff
@@ -476,6 +476,17 @@ func ByIds(ids []string) db.Q {
 	})
 }
 
+// FindByJasperCredentialsID finds a host with the given Jasper credentials ID.
+func FindOneByJasperCredentialsID(id string) (*Host, error) {
+	h := &Host{}
+	query := bson.M{JasperCredentialsIDKey: id}
+	err := db.FindOne(Collection, query, db.NoProjection, db.NoSort, h)
+	if adb.ResultsNotFound(err) {
+		return nil, err
+	}
+	return h, err
```

Comment:
wrap errors.

Reply:
I feel like these APIs that can return nil, nil are always confusing for users and I don't have a great solution.

Reply:
Wrapped the error and rewrote this slightly for clarity.

This shouldn't return nil, nil - it always returns the error if there is one (unless anser could return nil, nil from `db.FindOne()`).